### DISCUSSION TITLE
No person object in <mg-person> component of pnp modern search results template

### DISCUSSION THREAD

**Discussion by SinghGurd** [microsoft-search/pnp-modern-search#2199] [Q&A] [answered]
I am trying to customise the template used for people search results in PnP modern search results webpart. As indicated in below code snippet, I am using the <mgt-person> graph component and then trying to use the template element to display person properties. However, within template element, there is no person object and  {{JSONstringify .}} prints out the search result item. {{person.displayName}} is always null.

`<mgt-person user-id="{{item.AADObjectID}}" person-card="hover" view="threeLines">
                     <template>
                       <div>
                           Hello, my name is: {{person.displayName}}
                           {{JSONstringify .}}
                       </div>
                    </template>
                    <template data-type="no-data">
                        <h2>Nothing found</h2>
                    </template>
                </mgt-person>`

**Comment by wobba** [comment, marked as answer]
If you switch to the debug template, then you can look at the JSON for which objects/properties are available to you.

**Reply by SinghGurd** [reply]
Thanks for the response @wobba...I have had no luck. Essentially, when I use the below code in [mgt.dev](https://mgt.dev/?path=/story/components-mgt-person--person&source=docs), it prints out the personImage. However, when exact same code is used in a custom layout that I have stored in a HTML file stored in SharePoint library, it prints nothing.

`<mgt-person person-query="me">
    <template data-type="default">
        {{personImage}}
    </template>
</mgt-person>`

Custom layout template
`<content id="data-content">
            <ul class="template--defaultList">
                {{#each @root.data.items as |item|}}               
                <li tabindex="0">
                    <mgt-person person-query="me">
                        <template data-type="default">
                            Img: {{personImage}}
                            Person: {{person}}
                            Result item: {{item}}
                        </template>
                    </mgt-person>
                 </li>
                {{/each}}
            </ul>
        </div>
</content>`

Output of custom layout is as below. 
Img: Person: Result item: [object Object]

**Reply by FranckyC** [reply]
This one is a little bit tricky. It is because you need to escape the value `{{personImage}}` expression with "\\" to get it work. This is because your are in a Handlebars context and the default `{{ }}` is interpreted in the Handlebars context first, not MGT context. As an example, in the "People" layout we use it this way:

```
  <mgt-person user-id="{{getUserEmail (slot item @root.slots.PersonQuery)}}" person-card="hover">
      <template>
          <pnp-persona 
              data-image-url="\{{personImage}}" 
              data-fields-configuration="{{JSONstringify @root.properties.layoutProperties.peopleFields}}" 
              data-item="{{JSONstringify item}}" 
              data-persona-size="{{@root.properties.layoutProperties.personaSize}}"
              data-theme-variant="{{JSONstringify @root.theme}}"
              data-instance-id="{{@root.instanceId}}"
              data-context="{{JSONstringify @root}}">
          </pnp-persona>
      </template>
      <template data-type="person-card">
          <mgt-person-card inherit-details>
          </mgt-person-card>
      </template>
  </mgt-person>
```

**Reply by SinghGurd** [reply]
A BIG thank you @FranckyC...This has worked for me. Could I please run one more thing past you? How to use it in an if expression? E.g., the below code results in error.

`{{#if \(person.companyName) }}<p>\{{person.companyName}}</p>{{/if}}`

**Reply by mpsalian** [reply]
I am trying to add and Custom class for <pnp-persona  but unable to do so , Could you guide me here ?


###   <mgt-person user-id="{{getUserEmail (slot item @root.slots.PersonQuery)}}" person-card="hover">
                                            <template>
                                                **<pnp-persona** 
                                                    data-image-url="\{{personImage}}" 
                                                    data-fields-configuration="{{JSONstringify @root.properties.layoutProperties.peopleFields}}" 
                                                    data-item="{{JSONstringify item}}" 
                                                    data-persona-size="{{@root.properties.layoutProperties.personaSize}}"
                                                    data-theme-variant="{{JSONstringify @root.theme}}"
                                                    data-instance-id="{{@root.instanceId}}"
                                                    data-context="{{JSONstringify @root}}">
                                                </pnp-persona>
                                            </template>
                                            <template data-type="person-card">
                                                <mgt-person-card inherit-details>
                                                </mgt-person-card>
                                            </template>
                                        </mgt-person>



this  is my code

**Reply by wobba** [reply]
@mpsalian better if you start a separate discussion for your scenario.
